pkg/controller/v1beta1/inferenceservice/utils/utils_test.go

```diff
@@ -1223,8 +1223,6 @@ func TestMergeRuntimeContainers(t *testing.T) {
 				Name:  "kserve-container",
 				Image: "default-image",
 				Args: []string{
-					"--foo=bar",
```

Comment:
Thanks @pvaneck, this is a tricky one. I actually wasn't sure how best to handle args so thought we may as well follow the Kubernetes merge rules for that field (replace rather than combine). 

The trouble is that it's kind of situation dependent... i.e. the syntax/semantics of a particular model server's args and the reason that they are being set in the SR. Even in the case of the referenced test failure, the goal is to set a different http port and it will end up adding two such parameters rather than replacing the [existing one](https://github.com/kserve/kserve/blob/89d56de823db57fa40d2458fd0d8178e7a578f28/manifests/charts/templates/clusterservingruntimes.yaml#L127).

But I can see that here that out of the two options, appending is more likely to be the desired behaviour, so I'll make an update to restore that.